BlueDBOnDisk/src/main/java/org/bluedb/disk/backup/BackupManager.java

```diff
@@ -163,8 +165,15 @@ private void copyDataFileStraightOver(ReadWriteSegment<?> segment, Path tempFold
 		try (BlueReadLock<Path> lock = segment.getReadLockFor(groupingNumber)) {
 			Path src = lock.getKey();
 			Path dst = translatePath(dbPath, tempFolder, src);
-			// Change this
-			FileUtils.copyFileWithoutLock(src, dst);  // already have read lock on src, shouldn't need write lock on dst
+
+			try(BlueObjectOutput<BlueEntity<?>> output = BlueObjectOutput.createWithoutLockOrSerializer(dst, this.encryptionService, true)) {
+				try(BlueObjectInput<?> input = segment.getObjectInputFor(groupingNumber)) {
+					while(input.hasNext()) {
+						input.nextWithoutDeserializing();
+						output.writeBytesAndForceSkipEncryption(input.getLastUnencryptedBytes());
+					}

```

Comment:
I'll look into this, in retrospect `forceWriteUnencrypted` isn't really accurate, it is just initializing the metadata as unencrypted, overwriting the encryption setting. I could use that variable when writing however to bypass any other setting, though it may be unnecessary where the correct write method should be called in the first place. I'll update with what I decide, I'll think on it for a few minutes to be sure whatever solution I go with is solid. I am leaning towards your suggestion of updating `writeAll`, maybe following the same pattern as the other `writeBytes` methods and wrapping it with two method names that reflect whether encryption is allowed.

Reply:
I've pushed a change here, I went ahead and made the boolean in the private constructor drive a class level field that force skips all encryption for writes. I'm still digging in a bit more to see if I can come up with a cleaner solution than this but I am pretty confident this will work if we can't come up with something cleaner.

Reply:
Cool, I think this is a good as we're going to get it. We'll have to rely on test coverage to make sure devs aren't passing in encrypted bytes when they shouldn't and stuff like that.